src/flagStore.ts

```diff
@@ -142,21 +167,82 @@ export class FlagStore {
 					try {
 						flag = await this.api.getFeatureFlag(this.config.project, key, this.config.env);
 						this.flagMetadata[key] = flag;
-					} catch (e) {
-						console.error(`Could not retrieve feature flag metadata for ${key}: ${e}`);
-						reject(e);
-						return;
+					} catch (err) {
+						let errMsg = 'Error retrieving Flags';
+						if (err.statusCode === 401) {
+							errMsg = 'Unauthorized';
+						} else if (
+							err.statusCode === 404 ||
+							(err.statusCode === 400 && err.message.includes('Unknown environment key'))
+						) {
+							errMsg = 'Configured environment does not exist.';
+						}
+						window.showErrorMessage(`[LaunchDarkly] ${errMsg}`);
 					}
 				}
-
-				resolve({ flag, config: res });
+				const retFlag = await this.mergeFlag(flag, res);
+				resolve(retFlag);
 			});
 		});
 	}
 
-	allFlags(): Promise<FlagConfiguration[]> {
+	async allFlags(): Promise<Record<string, FeatureFlag>> {
+		await this.ldClient;
+		const flagDeepClone = JSON.parse(JSON.stringify(this.flagMetadata));
```

Comment:
Same as the other comment - we shouldn't need to `JSON.parse(JSON.stringify())` anything

Reply:
Without the JSON stringify I'm getting an error like this:

```rejected promise not handled within 1 second: TypeError: Cannot read property 'dano' of undefined
extensionHostProcess.js:976
stack trace: TypeError: Cannot read property 'dano' of undefined
	at FlagStore.mergeFlag (/Users/danielobrien/Projects/ld-vscode/dist/extension.js:52184:62)
	at FlagStore.<anonymous> (/Users/danielobrien/Projects/ld-vscode/dist/extension.js:52154:44)
	at Generator.next (<anonymous>)
	at /Users/danielobrien/Projects/ld-vscode/dist/extension.js:51967:71```

line 52184 in extension.js is:
```                const retFlag = yield this.mergeFlag(flag, res);``` which is around line 183 https://github.com/launchdarkly/ld-vscode/blob/storeRefactor/src/flagStore.ts#L183

Reply:
This appears to be due to some incorrect downstream assumptions of what `flagMetadata` (the FlagMap) type should look like. The roundabout stringify -> parse forces loss of type data - returning a plain js object.

The `mergeFlag` and `mergeAll` methods look pretty suspect in how they reassign values to an object. Would recommend doing some debugging in that part to make sure the flag data is being merged as expected

Reply:
We might have briefly discussed this idea in a call, but I never wrote it down -- one way to avoid the complexity of merging flag configs would be to only use the API results for storing the global flag metadata, without attempting to merge environment configs. This approach should be ok because the stream guarantees up-to-date configuration data.

Reply:
the reason I had to merge it was for things like the Treeview, to give them a clean API to work with. I was able to remove the Stringify! Let me know what you think.